src/confirmer.js

```diff
@@ -205,7 +205,7 @@ class Confirmer {
    */
   static resolve(result) {
     if (result instanceof Confirmer) { return result; }
-    let newConfirmer = new Confirmer(() => {});
+    let newConfirmer = Object.create(Confirmer.prototype);
```

Comment:
what does Object.create get you over `new`?

Reply:
`Object.create()` will create an instance of the object *without* calling the constructor. In this case it will skip assigning `this._proimise` because the factory function here immediately assigns it in the next line.

Learned this trick from https://stackoverflow.com/a/50856428